lambdas/preview/index.py

```diff
@@ -163,13 +168,61 @@ def extract_vcf(file_):
                 break
 
     # pylint: disable=bad-continuation
-    html = ("<div>"
+    html = ('<div>'
                 "<div data-type='meta'>{}</div>"
                 "<div data-type='header'>{}</div>"
                 "<div data-type='data'>{}</div>"
-            "</div>").format(
-                ''.join(meta) + '<br />',
-                ''.join(header) + '<br />',
+            '</div>').format(
+                ''.join(meta),
+                ''.join(header),
                 ''.join(data))
 
     return html, {}
+
+def extract_txt(file_):
+    """
+    Display first and last lines of file.
+    Uses UNIX utils for speed and convenience (no easy way to tail in Python).
+
+    Args:
+        file_ - file-like
+
+    Throws:
+        CalledProcessError
+    """
+    available = 0
+    ellipsis = False
+    # find out how many lines <= MAX_LINES are available
+    # we need to know this so that head and tail don't overlap
+    # loop finishes in constant time
+    with open(file_.name, 'rb') as txt:
+        for index, line in enumerate(txt, start=1):
+            available = index
+            if available > MAX_LINES:
+                ellipsis = True
+                break
+    # if maxlines == 0, everything still works
+    available = min(available, MAX_LINES)
+    # pylint should never flag for 3.6 but it does
+    headlines = int(available/2) # pylint: disable=old-division
+    taillines = available - headlines
+
+    head = subprocess.check_output(['head', file_.name, '-n', str(headlines)])
+    tail = subprocess.check_output(['tail', file_.name, '-n', str(taillines)])
+    html_buffer = [[], []]
+    for i, lines in enumerate([head, tail]):
+        for line in lines.splitlines():
+            line = line.strip().decode('unicode_escape')
+            html_buffer[i].append(f'<p>{line}</p>')
+
+    # pylint: disable=bad-continuation
+    html = ('<div>'
```

Comment:
Should, pick a consistent quote to use. Looks like maybe double quotes for python strings and single quotes withint the strings for html values?

Reply:
single quote is PEP-8 default but when i have nested quotes then double quote becomes necessary

Reply:
Actually PEP-8 and Google Style don't default to singles. I'd just swap them all to double so that the single quotes inside don't have to be escaped

> String Quotes
> In Python, single-quoted strings and double-quoted strings are the same. This PEP does not make a recommendation for this. Pick a rule and stick to it. When a string contains single or double quote characters, however, use the other one to avoid backslashes in the string. It improves readability.
https://www.python.org/dev/peps/pep-0008/#string-quotes

Reply:
we're returning JSON now so doesn't matter, but noted for future